src/bp-groups/bp-groups-template.php

```diff
@@ -4368,6 +4368,30 @@ function bp_get_group_join_button( $group = false ) {
 		return bp_get_button( apply_filters( 'bp_get_group_join_button', $button, $group ) );
 	}
 
+if ( ! function_exists( 'bb_group_btn_label_text_make_grammatical' ) ) {
+	/**
+	 * Make Correct grammaticle group label.
+	 *
+	 * @since BuddyBoss X.X.X
+	 *
+	 * @param string $full_text  Label of Group User type.
+	 * @param int    $group_id   Group Id
+	 * @param string $label_text Label of Group User type
+	 * @return string
+	 */
+	function bb_group_btn_label_text_make_grammatical( $full_text, $group_id, $label_text ) {
+		$is_an = apply_filters( 'bb_group_btn_label_text_make_grammatical_regex', preg_match( "/^[aeiou]/i", trim( strtolower( $label_text ) ) ) );
+
+		if ( $is_an ) {
+			$full_text = apply_filters( 'bb_group_btn_label_text_make_grammatical', sprintf( __( "You're an %s", 'buddyboss' ), $label_text ), $group_id, $label_text );
+		} else {
+			$full_text = apply_filters( 'bb_group_btn_label_text_make_grammatical', sprintf( __( "You're a %s", 'buddyboss' ), $label_text ), $group_id, $label_text );
+		}
+
+		return $full_text;
```

Comment:
@meetmakadiya apply_filters should be added to this line. why we adding that in if...else.